video-app/src/app/store/videos/video.actions.ts

```diff
@@ -0,0 +1,47 @@
+import { createAction, props } from '@ngrx/store';
+import { AddVideoResult } from '@core/models/addVideoResult';
+
+export const addYouTubeVideo = createAction(
+  '[Video] - Add YouTube Video',
+  props<{ videoId: string }>()
+);
+
+export const addYouTubeVideoSuccess = createAction(
+  '[Video] - Add YouTube Video Success',
+  props<AddVideoResult>()
+);
+
+export const addYouTubeVideoFail = createAction(
+  '[Video] - Add YouTube Video Fail',
+  props<{ fail: boolean }>()

```

Comment:
ten argument jest niepotrzebny, akcja to `Fail` więc `fail` będzie zawsze jako true